web/access_control.go

```diff
@@ -131,15 +118,17 @@ func (s *AutograderService) isCourseCreator(courseID, userID uint64) bool {
 	return course.GetCourseCreatorID() == userID
 }
 
+// TODO(vera): these two methods can be repurposed to return course-related scm client for most methods
```

Comment:
We should probably separate this into two methods. One for course-specific scm and one for personal access token-based scm for accepting invitations. That way, we could probably returns a more narrow interface for the access token scm with only the relevant methods. That is, we should probably define a new interface for it.

Checked: This method may no longer be necessary, since the current user is only used for access control in `UpdateCourse`.